<launch>
  <param name="use_sim_time" value="true"/>
  <arg name="bagfile"/>
  <arg name="doalign" default="0"/>
  <arg name="worldmap" default=""/>

  <node name="visualizer" pkg="rviz" type="rviz"/>
  <param name="current_wifi_playback" value="$(arg bagfile).bag"/>

  <include file="$(find hector_slam_launch)/launch/mapping_box.launch"/>
  <node pkg="tf" type="static_transform_publisher" name="laser_broadcaster" args="0 0 .1 0 0 0 base_link laser 100" />
  <node name="pose_estimation" pkg="hector_pose_estimation" type="pose_estimation" output="screen">
    <remap from="poseupdate" to="dflksdjflksdjf" />
  </node>

  <node name="replay" pkg="rosbag" type="play" args="--clock $(arg bagfile).bag -r 10" required="true"/>

  <!-- the worldmap argument can be used to give a worldmap.yaml file previously saved with the map_saver node,
       at the same time a $bagfile.yaml file will be searched and used to publish the /map -> /world tf. Thus
       "aligning" the currently played back /map to the /world frame.
  -->
  <group if="$(arg doalign)">
    <node name="world_map_server" pkg="map_server" type="map_server" args="$(arg worldmap)" required="true">
      <param name="frame_id" value="world"/>
      <remap from="map" to="world"/>
    </node>

    <node name="align_map_server" pkg="map_server" type="map_server" args="$(arg bagfile).yaml" required="true">
      <remap from="map" to="align"/>
    </node>

    <node name="align_to_world" pkg="mapstitch" type="ros_mapstitch" required="true" output="screen">
      <param name="debug" value="false"/>
      <param name="max_distance" value="3."/>
    </node>

    <node name="rssi_csv_export" pkg="rssimap" type="save_rssi.sh" required="true"/>
  </group>

  <node name="rssi_pose" pkg="rssimap" type="rssi-pose.py" required="true"/>
</launch>
